okhttp/src/main/java/okhttp3/Interceptor.java

```diff
@@ -26,6 +26,7 @@
  */
 public interface Interceptor {
   Response intercept(Chain chain) throws IOException;
+  void cleanup();
```

Comment:
This will create a huge amount of pain for users.  There are lots of implementors of Interceptors.  So there is a massive barrier for this change to start.

Also cleanup() doesn't fit the lifecycle of an interceptor which may be used for millions of concurrent requests.  Interceptors already have a chance to cleanup as part of their internal logic in intercept.

Reply:
Sounds good, this was one of my concerns with this change.